### ISSUE TITLE
ERROR: zero-length delimited identifier at or near """"

### ISSUE THREAD

**Issue by KiARC** [prisma/prisma#22968]
### Bug description

Anything I try to do with prisma (migrations, pushes, etc) fails with the following error:

```
Error: Schema engine error:
ERROR: zero-length delimited identifier at or near """"
```

This persists even after nuking and re-initializing the entire database.

### How to reproduce

1. Start a NextJS project, add Prisma, and create a postgres DB (mine is running in a container, if that matters)
2. Write a schema.prisma file, I'll include mine below
3. Run `yarn run prisma migrate dev --name init`, `yarn run prisma db push`, or another similar command

### Expected behavior

The DB is adjusted to fit the schema, with new tables created to match the config file.

### Prisma information

```
// schema.prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Submission {
  id        Int      @id @unique @default(autoincrement())
  ip        String
  content   String
  notes     String
  tags      String[]
  timestamp DateTime @default(now())
}

model Post {
  id        Int      @id @unique @default(autoincrement())
  content   String
  notes     String
  tags      String[]
  upvotes   Int
  downvotes Int
  revoked   Boolean  @default(false)
  timestamp DateTime @default(now())
  Report    Report[]
}

model Report {
  id           Int            @id @unique @default(autoincrement())
  post         Post          @relation(fields: [postId], references: [id])
  reason       String
  closed       Boolean        @default(false)
  timestamp    DateTime       @default(now())
  postId       Int
  ReportAction ReportAction[]
}

model ReportAction {
  id        Int      @id @unique @default(autoincrement())
  report    Report   @relation(fields: [reportId], references: [id])
  decision  Boolean
  timestamp DateTime @default(now())
  reportId  Int
}

model NewsPost {
  id        Int      @id @unique @default(autoincrement())
  content   String
  author    String
  timestamp DateTime @default(now())
}
```

(there is no client code involved, this error happens with migrations and other CLI operations)

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: v20.11.0

### Prisma Version

```
prisma                  : 5.9.1
@prisma/client          : 5.9.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.9.0-32.23fdc5965b1e05fc54e5f26ed3de66776b93de64
Default Engines Hash    : 23fdc5965b1e05fc54e5f26ed3de66776b93de64
Studio                  : 0.497.0
```

**Comment by KiARC** [comment]
I'll also note that this started happening seemingly randomly. No changes were made to the schema; I had an error come out of a query (same error as above), tried to migrate to see if that would fix it, and got this.

**Comment by janpio** [comment]
Does this happen with all databases or just this specific one?
Does it happen when you try to use a MySQL database? (PlanetScale.com offers free ones to play around with)

**Comment by KiARC** [comment]
I haven't tested anything besides postgres, but I will check tonight when I have a chance. It does happen on a brand new Postgres DB with minimal configuration (only the username, password and database name being set, all to "postgres"), and reverting commits in my repo that affect the schema doesn't seem to do anything. Deleting migrations also has not fixed the issue.

**Comment by vanyle** [comment]
I had the same issue. On the postgresql side, I had the following error:
```
ERROR:  zero-length delimited identifier at or near """" at character 19
STATEMENT:  SET search_path = ""
```

This means that Prisma is confused about the database you are using. Check if your `.env` file is properly loaded and that your database connection url is properly formatted.

**Comment by janpio** [comment]
Do I assume correctly that your connection string was missing the database name in some way, and it was not caught by Prisma @vanyle?

**Comment by vanyle** [comment]
Yes, what happen for me was that it was not properly loaded from my `.env` file on windows.

`.env` :
```
DB_URL="postgresql://postgres:postgres@localhost:5432/postgres?schema=public"
```

`schema.prisma` :

```
...
datasource db {
  provider = "postgresql"
  url      = env("DB_URL")
}
...
```

Because I used all the default values (`postgres`) for user, password and database, there were no authentification issues, but for some reason, prisma had trouble as the environment variable for the url was `DB_URL` and not `DATABASE_URL` like inside the documentation which made `prisma db push` fail.

Setting it back to `DATABASE_URL` seems to work (I changed the username, password and database name inside postgres to make sure that everything actually works)

**Comment by vanyle** [comment]
This line: https://github.com/prisma/prisma/blob/main/packages/bundle-size/da-workers-pg/index.js#L8 looks strange.
How do we know that the connection string is actually inside the `DATABASE_URL` environment variable?

**Comment by KiARC** [comment]
My `.env` always had `DATABASE_URL` but I'm not having the issue anymore after porting the project to a new framework (NextJS -> Remix) so perhaps I had messed something up in my project setup.

**Comment by janpio** [comment]
> [This line](https://github.com/prisma/prisma/blob/main/packages/bundle-size/da-workers-pg/index.js?rgh-link-date=2024-02-14T16%3A45%3A20Z#L8) looks strange. How do we know that the connection string is actually inside the `DATABASE_URL` environment variable?

The code you link is basically a test project we use internally, so this does not mean that Prisma only can handle `DATABAES_URL` - it is just the one we use in that example. You can use whatever name for the env var, as long as it is `env("FOO")` in the schema file and then read make sure the connection string is available in `FOO` environment variable things should work.

I would be really interested in reproducing this problem. In theory Prisma _should_ catch empty database names and give a much better, helpful error message.

**Comment by vikyw89** [comment]
![image](https://github.com/prisma/prisma/assets/112059651/5119b311-e3a2-4798-ba28-fa935fa5703e)
```
generator client {
  provider             = "prisma-client-py"
  interface            = "asyncio"
  recursive_type_depth = 5
}

datasource db {
  provider = "postgresql"
  url      = env("PRISMA_DATABASE_URL")
}
```

happened to me today multiple times, haven't experienced this before

**Comment by drautu** [comment]
I encounter the same issue and on my side I solved it by unset environment var `DATABASE_URL` before running `npx prisma db migrate/push `
I didn't research where exactly it is set that env var when I open my bash terminal in vscode.

Hope it helps!

**Comment by entvia** [comment]
As @drautu mentioned, `unset DATABASE_URL` fixes the problem. I think it is related to the fact that in my case `echo $DATABASE_URL` showed a url with empty `schema` parameter. I don't know why it happened though.
